src/daemon/mod.rs

```diff
@@ -438,10 +437,15 @@ pub(super) async fn start(
         info!("Imported snapshot in: {}s", stopwatch.elapsed().as_secs());
     }
 
-    if config.client.snapshot {
-        if let Some(validate_height) = config.client.snapshot_height {
-            ensure_params_downloaded().await?;
-            validate_chain(&state_manager, validate_height).await?;
+    if let (true, Some(validate_from)) = (config.client.snapshot, config.client.snapshot_height) {
+        /// We've been provided a snapshot and asked to validate it
+        ensure_params_downloaded().await?;
+        let current_height = state_manager.chain_store().heaviest_tipset().epoch();
+        assert!(current_height.is_positive());
+        match validate_from.is_negative() {
```

Comment:
What is wrong with a simple `if-else`?

Reply:
Same as here:
https://github.com/ChainSafe/forest/pull/2893#discussion_r1220358284